src/pings.rs

```diff
@@ -0,0 +1,58 @@
+use crate::User;
+use core::fmt;
+
+#[derive(Debug)]
+pub enum Ping {
+    JobCancelled { id: u16, reason: String },
+    JobFinished { id: u16 },
+    JobStarted { id: u16 },
+}
+
+impl fmt::Display for Ping {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        use crate::Ping::*;
+        match self {
+            JobCancelled { id, reason } => write!(f, "Job {} cancelled: {}", id, reason),
+            JobStarted { id } => write!(f, "Job {} started", id),
+            JobFinished { id } => write!(f, "Job {} finished", id),
+        }
+    }
+}
+
+impl Ping {
+    //find a better way to do this
+    fn get_route_uuid(&self) -> &str {
+        use crate::Ping::*;
+        match self {
+            JobCancelled { .. } => "97f166a0-d9f9-4888-b8ad-b09656f19330",
+            JobStarted { .. } => "f810410a-0dcd-4271-8b75-b6e6ce5ead7e",
+            JobFinished { .. } => "aab2a0c6-001c-455a-956f-10c8b685fb87",
```

Comment:
I'm extremely wary of hardcoding magic numbers into applications. What the hell is this, and can it be broken out into a config file in a way that is human readable?

Reply:
I agree, but this is a temporary measure until I find a not terrible way to put this in a config file.

Reply:
There's another JSON file in the app somewhere, right? Could you use that?

Alternatively: https://crates.io/crates/config